### PR TITLE
Merge 2.15.x up into 3.0.x

### PR DESCRIPTION
**Pull request by greg0ire**
As you predicted @derrabus, that one was quite painful. I used https://github.com/mhagger/git-imerge, but still.

Note that I'm not even merging the tip of 2.15.x, but the commit right before we bump Psalm to 5.5.0, so there will be a need for another merge up after this one. Just thought it would be best to deal with the Psalm bump separately.

Cc @mpdude

### CODE REVIEW

**Comment by greg0ire** [comment]
Seems it fails with DBAL 4 :disappointed: 
Somebody must have introduced some deprecated calls.

File: tests/Doctrine/Tests/ORM/Tools/Console/Command/SchemaTool/UpdateCommandTest.php [hunk 1]
```diff
@@ -26,4 +26,29 @@ public function testItPrintsTheSql(): void
...
+        yield 'default_name' => [
+            null,
+            '[WARNING] Not passing the "--complete" option to "orm:schema-tool:update" is deprecated',
+        ];
+
+        yield 'custom_name' => [
+            'doctrine:schema:update',
+            '[WARNING] Not passing the "--complete" option to "doctrine:schema:update" is deprecated',
+        ];
+    }
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Tools/Console/Command/SchemaTool/UpdateCommandTest.php:57]: Ah that must be the culprit code :mag:

Comment (greg0ire) [reply tests/Doctrine/Tests/ORM/Tools/Console/Command/SchemaTool/UpdateCommandTest.php:57]: Introduced in #10443

**Comment by mpdude** [comment]
Looks good from my POV, at least from what I can tell off the top of my head.

Keep up the good work!

Resolving merge conflicts like these is the hard work maintainers do in the background, without hardly anyone noticing.

**Review by derrabus — APPROVED** [review]

**Pull request merged**